src/sources/soundsourcemp3.cpp

```diff
@@ -0,0 +1,589 @@
+#include "sources/soundsourcemp3.h"
+
+#include "util/math.h"
+
+#include <id3tag.h>
+
+namespace Mixxx {
+
+namespace {
+
+// In the worst case up to 29 MP3 frames need to be prefetched
+// for accurate seeking:
+// http://www.mars.org/mailman/public/mad-dev/2002-May/000634.html
+const SINT kSeekFramePrefetchCount = 29;
+
+const CSAMPLE kMadScale = AudioSource::kSampleValuePeak
+        / CSAMPLE(MAD_F_ONE);
+
+inline CSAMPLE madScale(mad_fixed_t sample) {
+    return sample * kMadScale;
+
+} // anonymous namespace
+
+// Optimization: Reserve initial capacity for seek frame list
+const SINT kMinutesPerFile = 10; // enough for the majority of files (tunable)
+const SINT kSecondsPerMinute = 60; // fixed
+const SINT kMaxMp3FramesPerSecond = 39; // fixed: 1 MP3 frame = 26 ms -> ~ 1000 / 26
+const SINT kSeekFrameListCapacity = kMinutesPerFile
+        * kSecondsPerMinute * kMaxMp3FramesPerSecond;
+
+void logFrameHeader(QDebug logger, const mad_header& madHeader) {
+    logger << "MP3 frame header |"
+            << "layer:" << madHeader.layer
+            << "mode:" << madHeader.mode
+            << "#channels:" << MAD_NCHANNELS(&madHeader)
+            << "#samples:" << MAD_NSBSAMPLES(&madHeader)
+            << "bitrate:" << madHeader.bitrate
+            << "samplerate:" << madHeader.samplerate
+            << "flags:" << madHeader.flags;
```

Comment:
Valid point! I was simply too lazy to look this up myself ;) Commit will follow.